src/secondary_thermal.cpp

```diff
@@ -32,25 +33,29 @@ CoherentElasticAE::CoherentElasticAE(const CoherentElasticXS& xs) : xs_ {xs} {}
 void CoherentElasticAE::sample(
   double E_in, double& E_out, double& mu, uint64_t* seed) const
 {
-  // Get index and interpolation factor for elastic grid
-  int i;
-  double f;
+  // Energy doesn't change in elastic scattering (ENDF-102, Eq. 7-1)
+  E_out = E_in;
+
   const auto& energies {xs_.bragg_edges()};
-  get_energy_index(energies, E_in, i, f);
+
+  if (E_in < energies.front()) {
+    // XS is zero for E < energies[0] -> no scatter
+    mu = 1.0;
+    return;
+  }
```

Comment:
Sorry, one more thing (I thought I had made this comment before but I must have forgotten). This should actually never happen; if the cross section for coherent elastic is zero, we shouldn't be sampling for the energy/angle. I would change this to:
```suggestion
  Expects(E_in >= energies.front());
```
with a `#include <gsl/gsl-lite.hpp>` up top